test/unit/server/soroban/simulate_transaction_test.js

```diff
@@ -275,6 +356,49 @@ async function invokeSimulationResponseWithRestoration(address) {
   };
 }
 
+async function invokeSimulationResponseWithStateChanges(address) {
+  return {
+    ...(await invokeSimulationResponse(address)),
+    stateChanges: [
+      {
+        type: 2,
+        key: "example-key-2",
+        before: new xdr.LedgerEntry({
+          lastModifiedLedgerSeq: 0,
+          data: new xdr.LedgerEntryData(),
+          ext: new xdr.LedgerEntryExt(),
+        }).toXDR("base64"),
+        after: new xdr.LedgerEntry({
+          lastModifiedLedgerSeq: 0,
+          data: new xdr.LedgerEntryData(),
+          ext: new xdr.LedgerEntryExt(),
+        }).toXDR("base64"),
+      },
+      {
+        type: 1,
+        key: "example-key-1",
+        before: null,
+        after: new xdr.LedgerEntry({
+          lastModifiedLedgerSeq: 0,
+          data: new xdr.LedgerEntryData(),
+          ext: new xdr.LedgerEntryExt(),
+        }).toXDR("base64"),
+      },
+      {
+        type: 3,
+        key: "example-key-3",
```

Comment:
wait shouldn't this fail to decode to a `LedgerKey`?

Reply:
Yes I forgot to update it, it did not fail because it was same string at both places (actual vs expected)